crowbar_framework/app/models/service_object.rb

```diff
@@ -1540,6 +1524,79 @@ def wait_for_chef_daemons(node_list)
 
   private
 
+  THREAD_POOL_SIZE = 20
+
+  def release_chef_locks(locks)
+    return if locks.empty?
+
+    queue = Queue.new
+    locks.each { |l| queue.push l }
+
+    workers = (1...THREAD_POOL_SIZE).map do
```

Comment:
> Still I do not get why to do that. The locks are release into a thread, but the main thread is still blocked in the .join(), so there is not performance improvement.
> 
> Why not to collect that in the main thread then?

@vuntz may now the intent of this, tbh in my tests there doesnt seem to be a performance improvement from this patch, Im guessing it may be helpful in certain conditions or its only a middle step to introduce further improvements down the line?